x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/eql/build_alert_group_from_sequence.ts

```diff
@@ -38,25 +84,27 @@ import type {
  * @param sequence The raw ES documents that make up the sequence
  * @param completeRule object representing the rule that found the sequence
  */
-export const buildAlertGroupFromSequence = (
-  ruleExecutionLogger: IRuleExecutionLogForExecutors,
-  sequence: EqlSequence<SignalSource>,
-  completeRule: CompleteRule<RuleParams>,
-  mergeStrategy: ConfigType['alertMergeStrategy'],
-  spaceId: string | null | undefined,
-  buildReasonMessage: BuildReasonMessage,
-  indicesToQuery: string[],
-  alertTimestampOverride: Date | undefined,
-  publicBaseUrl?: string,
-  intendedTimestamp?: Date
-): Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest | EqlShellFieldsLatest>> => {
+export const buildAlertGroupFromSequence = ({
+  ruleExecutionLogger,
+  sequence,
+  completeRule,
+  mergeStrategy,
+  spaceId,
+  buildReasonMessage,
+  indicesToQuery,
+  alertTimestampOverride,
+  publicBaseUrl,
+  intendedTimestamp,
+}: BuildAlertGroupFromSequence): BuildAlertGroupFromSequenceReturnType => {
```

Comment:
What's the reason for changing the return type of this function? We don't add subAlerts to the shell alert yet, so it seems to just add complexity with the optional types buried in this new return type.

Reply:
Fixed here: https://github.com/elastic/kibana/pull/189725/commits/3079b9eabcd2274281f1678704f1b09c5cc0e655